integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,288 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Optional, Union
+
+from haystack import DeserializationError, Pipeline, default_from_dict, default_to_dict, logging
+from haystack.components.embedders.types import TextEmbedder
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.joiners.document_joiner import JoinMode
+from haystack.core.serialization import component_from_dict, import_class_by_name
+from haystack.document_stores.types import FilterPolicy
+from haystack.lazy_imports import LazyImport
+
+from haystack_integrations.components.retrievers.opensearch import OpenSearchBM25Retriever, OpenSearchEmbeddingRetriever
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+
+logger = logging.getLogger(__name__)
+
+# Use LazyImport to conditionally import haystack super_component, which is only available in newer versions
+with LazyImport(
+    "To use the OpenSearchHybridRetriever you need a more recent version of haystack. Run 'pip install haystack-ai>=2.13.0'"  # noqa: E501
+) as haystack_imports:
+    from haystack import super_component
```

Comment:
Do we need this anymore if we are going to bump the required version of this package to haystack>=2.14.0?

Reply:
we can remove it yes, since now we have a dependency on the `TextEmbedder` Protocol which will be in `haystack>=2.14.0` we can remove it.

Reply:
> We could branch dependencies depending on which component you want to import (note: just an idea, would have to explore how to implement that)

Applying the lazyimport only when we try to import (OpenSearch) HybridRetriever could work.

But, unless there's an issue, I still would vote on making this dependent on the upcoming haystack release.

Reply:
In my opinion, the reason for the LazyImport and not requiring >=2.14.0 is still the same as when we discussed LazyImport for the OpenSearchHybridRetriever in the beginning: If we make any bug fixes or other changes in future to OpenSearchDocumentStore or one of the other retrievers, all users can benefit from those changes. Even if they can't/don't want to upgrade their Haystack version to >=2.14.0.

`haystack_imports.check()` is still in the implementation by the way [here](https://github.com/deepset-ai/haystack-core-integrations/pull/1701/files#diff-095ae23eb2de32c9f28370202fa6a9c147e6b73d5b1cfd6519995bf77ca3db38R20)

Reply:
reverted to 2.11 and only HybridRetriever for 2.14